autoPyTorch/pipeline/components/base_component.py

```diff
@@ -203,14 +219,22 @@ def check_requirements(self, X: Dict[str, Any], y: Any = None) -> None:
         if y is not None:
             warnings.warn("Provided y argument, yet only X is required")
 
-        for requirement in self._fit_requirements:
-            if requirement.name not in X.keys():
-                raise ValueError("To fit {}, expected fit dictionary to have '{}'"
-                                 " but got \n {}".format(self.__class__.__name__, requirement.name, list(X.keys())))
-            elif not isinstance(X[requirement.name], requirement.type):
-                raise TypeError("Expected {} to be instance of {} got {}".format(requirement.name,
-                                                                                 requirement.type,
-                                                                                 type(X[requirement.name])))
+        if self._fit_requirements:
+            for requirement in self._fit_requirements:
+                if requirement.name not in X.keys():
+                    raise ValueError("To fit {}, expected fit dictionary to have '{}'"
+                                     " but got \n {}".format(self.__class__.__name__, requirement.name, list(X.keys())))
+                else:
+                    UNSUPPORTED_TYPE = True
+                    for supported_type in requirement.supported_types:
+                        if isinstance(X[requirement.name], supported_type):
```

Comment:
you can get rid of this for loop and variable assignment like this:

`TYPE_SUPPORTED = isinstance(X[requirement.name], tuple(requirement.supported_types))`

since `isinstance` supports checking multiple types given as a tuple